### DISCUSSION TITLE
URL points back to client instead of sync server

### DISCUSSION THREAD

**Discussion by sutrotka** [TriliumNext/Trilium#2958] [Q&A] [answered]
When syncing my client to server i get this error message:

 "Sync server handshake failed, error: Sync server has member ID GIatWVsaoY8J which is also local. This usually happens when the sync client is (mis)configured to sync with itself (URL points back to client) instead of the correct sync server."

I've read other discussions about this exact thing to try and avoid annoying the support community but the solutions to them all require the op to "configure" their sync address. I'm very new to linux github and all of the above so I know this is a stupid question but please help lol. I want to sync my desktop trilium, with my laptop triulium both running on the same network which i assume to be the actual issue, not sure how to identify one to the other. 

Thanks in advance to anyone willing to help

**Comment by sigaloid** [comment, marked as answer]
The error does indeed occur when you set the sync URL to be its own URL. You only set up sync on the client side, not the server; the client is syncing *from* the server so it's the only one that needs to have sync configured. 

I bet you chose "I have a desktop instance set up already" in the setup which tells you to configure the sync on the other device instead of configuring it on the existing device. It can be a tad confusing, choosing "desktop" or "server", but your desktop would be the "server" in this case.

In this situation you'll probably want your desktop to serve as the "server" and the laptop to be a client (less changing IP addresses). 

**On the desktop simply install Trilium and locate the IP address of it. Run Trilium from the command line as well, so you can make sure the port is correct.** 

![image](https://user-images.githubusercontent.com/69441971/176975443-539e549c-ce45-4e47-b82b-022ca8fa7940.png)

**On the laptop, remove all of the data by clearing out `~/.local/share/trilium-data` (assuming your data is currently stored on the desktop! don't clear it from the laptop if that's your only copy).** 

**Then, go through the setup, and select "I have desktop instance already, and I want to set up sync with it".** 

**Enter the desktop's IP address (find this by typing `ip address` in your terminal) along with the port (37840 if you haven't changed it; we also found it above by starting Trilium in the command line).**

![image](https://user-images.githubusercontent.com/69441971/176975350-f42fa86d-77b7-4aa6-82d8-45c6c67376bc.png)
It will then start the syncing process and then any future changes on either side will sync between the two devices.

**Reply by sutrotka** [reply]
Ive verified the port on my desktop is correct, but when going to remove /.local/share/trilium-data I cant find it.  I downloaded the file straight from github then extracted to desktop.  Ive looked in every directory and every folder. So i opted to delete the extracted folder and redownload trilium, after the second trilium has finished downloading it has a (2) at the end.  I cant find the originals location for the life of me in terminal nor in the file explorer, and when i go to delete it so it will reset the app with a fresh download it remembers my previous setup somehow.  Pls point me in the right direction

**Reply by sigaloid** [reply]
It's not in /.local/share/trilium-data, it's in ~/.local/share/trilium-data. The `~` is a special symbol that means "my home directory". 

The file you download and extract from GitHub is the app, but the actual data is stored in the trilium-data directory. The app's data is not stored in the "trilium-linux-x64-0.52.4" folder. 

Run `rm -rf ~/.local/share/trilium-data` in your terminal.
